xbmc/guilib/GUIIncludes.cpp

```diff
@@ -513,13 +513,22 @@ std::string CGUIIncludes::ResolveConstant(const std::string &constant) const
   return StringUtils::Join(values, ",");
 }
 
-std::string CGUIIncludes::ResolveExpressions(const std::string &expression) const
+std::string CGUIIncludes::ResolveExpressions(const std::string &expression, const std::vector<std::string> &resolved /* = std::vector<std::string>() */) const
 {
   std::string work(expression);
   CGUIInfoLabel::ReplaceSpecialKeywordReferences(work, "EXP", [&](const std::string &str) -> std::string {
-    std::map<std::string, std::string>::const_iterator it = m_expressions.find(str);
-    if (it != m_expressions.end())
-      return it->second;
+    if (std::find(resolved.begin(), resolved.end(), str) != resolved.end())
+      CLog::Log(LOGWARNING, "Skin has a circular expression involving: %s", str.c_str());
```

Comment:
We should change the log level to error and also log the input string to make it easier for skinners to spot such a circular expression.